test/integration/smart-loan.test.ts

```diff
@@ -424,5 +430,170 @@ describe('Smart loan', () => {
+        },
+        {
+          symbol: 'AVAX',
+          value: MOCK_AVAX_PRICE
+        }
+      ]
+
+      await pool.initialize(fixedRatesCalculator.address, borrowersRegistry.address, ZERO, ZERO);
+      await pool.connect(depositor).deposit({value: toWei("1000")});
+    });
+
+    it("should deploy a smart loan", async () => {
+      loan = await (new SmartLoan__factory(owner).deploy());
+      loan.initialize(supportedAssets.address, exchange.address, pool.address);
+
+      wrappedLoan = WrapperBuilder
+        .mockLite(loan)
+        .using(
+          () => {
+            return {
+              prices: MOCK_PRICES,
+              timestamp: Date.now()
+            }
+          })
+
+      await wrappedLoan.authorizeProvider();
+    });
+
+    it("should fund a loan", async () => {
+      await wrappedLoan.fund({value: toWei("100")});
+    });
+
+    it("should borrow funds", async () => {
+      await wrappedLoan.borrow(toWei("300"));
+
+      expect(fromWei(await wrappedLoan.getTotalValue())).to.be.equal(400);
+      expect(fromWei(await wrappedLoan.getDebt())).to.be.closeTo(300, 0.1);
+      expect(await wrappedLoan.getSolvencyRatio()).to.be.closeTo(BigNumber.from(1333), 1);
+    });
+
+    it("should invest", async () => {
+
+
+      await wrappedLoan.invest(toBytes32('USD'), toWei("9000", usdTokenDecimalPlaces));
+      await wrappedLoan.invest(toBytes32('LINK'), toWei("10000", usdTokenDecimalPlaces));
+
+      let balances = await wrappedLoan.getAllAssetsBalances();
+
+      const currentUSDTokenBalance = balances[0];
+      const currentLINKTokenBalance = balances[1];
+      expect(currentUSDTokenBalance).to.be.equal(toWei("9000", usdTokenDecimalPlaces));
+      expect(currentLINKTokenBalance).to.be.equal(toWei("10000", usdTokenDecimalPlaces));
+    });
+
+    it("should fail a sellout attempt", async () => {
+
+      let UPDATED_MOCK_PRICES = MOCK_PRICES.map(
+        (token: any) => {
+          if (token.symbol == 'USD' || token.symbol == 'LINK') {
+            token.value = 0.6 * token.value;
+          }
+          return token;
+        }
+      );
+
+      let loanUpdated = WrapperBuilder
+        .mockLite(loan)
+        .using(
+          () => {
+            return {
+              prices: UPDATED_MOCK_PRICES,
+              timestamp: Date.now()
+            }
+          })
+
+      let loan_solvent = await loanUpdated.isSolvent();
+      let loan_below_sellout_treshold = await loanUpdated.isBelowSelloutTreshold();
+      expect(loan_solvent).to.be.false;
+      expect(loan_below_sellout_treshold).to.be.false;
+      await expect(loanUpdated.sellout()).to.be.revertedWith('Cannot sellout a solvent account');
+    });
+
+    it("should sellout a loan covering whole debt", async () => {
+      let UPDATED_MOCK_PRICES = MOCK_PRICES.map(
+        (token: any) => {
+          if (token.symbol == 'USD' || token.symbol == 'LINK') {
+            token.value = 0.7 * token.value;
```

Comment:
I don't understand one thing. In a previous test token prices are worth 0.6 of initial value and the loan is not a subject to sellout. In this example, tokens are worth 0.7 (more), and the loan solvency is below sellout threshold. How is it possible?